drivers/regulator/regulator_pca9420.c

```diff
@@ -271,23 +271,14 @@ static unsigned int regulator_pca9420_count_voltages(const struct device *dev)
 					       config->desc->num_ranges);
 }
 
-/**
- * Part of the extended regulator consumer API
- * Returns the supported voltage in uV for a given selector value
- */
-static int32_t regulator_pca9420_list_voltages(const struct device *dev,
-					       unsigned int selector)
+static int regulator_pca9420_list_voltage(const struct device *dev,
```

Comment:
Just a note here: this API (and several APIs changed in later commits) are based on the Linux regulator API: https://docs.kernel.org/driver-api/regulator.html#c.regulator_list_voltage. I'm aware Zephyr isn't Linux, but I wanted to highlight the original reasoning for the design here. I'm not opposed to the API change though.

Reply:
One option could be to return the value, but then all regulators need to implement the capability of listing voltages, even simple GPIO switches. It may actually make sense to specify the voltage on switches as well, after all they turn on/off a power rail that has a fixed voltage.

Let me think about this and come up with a proposal.

Reply:
Considering that the voltage domain is (-INT32_MAX, INT32_MAX), we have no other choice but to return the value by reference. Otherwise, we have no way to indicate whether the given index is valid or not. On Linux, only positive voltages are supported by the API, in our case both positive/negative are supported.